src/extended-keys.ts

```diff
@@ -1,41 +1,97 @@
-import { createDataView } from '@metamask/utils';
+import { assertExhaustive, createDataView } from '@metamask/utils';
 
 import { validateBIP44Depth } from './BIP44Node';
+import type { Network } from './constants';
 import { compressPublicKey, decompressPublicKey } from './curves/secp256k1';
 import { decodeBase58check, encodeBase58check, isValidBytesKey } from './utils';
 
 // https://github.com/bitcoin/bips/blob/274fa400d630ba757bec0c03b35ebe2345197108/bip-0032.mediawiki#Serialization_format
-export const PUBLIC_KEY_VERSION = 0x0488b21e;
-export const PRIVATE_KEY_VERSION = 0x0488ade4;
+const PUBLIC_KEY_VERSION = 0x0488b21e;
+const PRIVATE_KEY_VERSION = 0x0488ade4;
 
-export type ExtendedKeyVersion =
-  | typeof PUBLIC_KEY_VERSION
-  | typeof PRIVATE_KEY_VERSION;
+const TESTNET_PUBLIC_KEY_VERSION = 0x043587cf;
+const TESTNET_PRIVATE_KEY_VERSION = 0x04358394;
 
 /**
  * An extended public or private key. Contains either a public or private key,
  * depending on the version.
  */
 type ExtendedKeyLike = {
-  version: ExtendedKeyVersion;
   depth: number;
   parentFingerprint: number;
   index: number;
+  network: Network;
   chainCode: Uint8Array;
 };
 
 type ExtendedPublicKey = ExtendedKeyLike & {
-  version: typeof PUBLIC_KEY_VERSION;
+  type: 'public';
```

Comment:
Is it intentional for this to be a breaking change? Could we not include the version bytes still?

Reply:
This is not a public type, so this is not a breaking change.